arangod/RestHandler/RestAdminClusterHandler.cpp

```diff
@@ -0,0 +1,1563 @@
+
+  auto rootPath = arangodb::cluster::paths::root()->arango();
+  VPackBuffer<uint8_t> trx;
+  {
+    VPackBuilder builder(trx);
+    arangodb::agency::envelope<VPackBuilder>::create(builder)
+      .read()
+        .key(rootPath->supervision()->health()->str())
+        .key(rootPath->plan()->str())
+        .key(rootPath->current()->str())
+        .done()
+      .done();
+  }
+
+  return AsyncAgencyComm().sendWriteTransaction(20s, std::move(trx))
+    .thenValue([this, ctx = std::move(ctx)](AsyncAgencyCommResult &&result) mutable {
+
+      auto rootPath = arangodb::cluster::paths::root()->arango();
+      if (result.ok() && result.statusCode() == 200) {
+        VPackSlice agency = result.slice().at(0);
+
+        VPackSlice health = agency.get(rootPath->supervision()->health()->server(ctx->server)->status()->vec());
+        if (!health.isNone()) {
+
+          bool isResponsible = isServerResponsibleForSomething(ctx->server,
+            agency.get(rootPath->plan()->vec()), agency.get(rootPath->current()->vec()));
+
+          // if the server is still in the list, it was neither in plan nor in current
+          if (isResponsible) {
+
+            auto rootPath = arangodb::cluster::paths::root()->arango();
+            auto planVersionPath = rootPath->plan()->version();
+            // do a write transaction if server is no longer used
+            VPackBuffer<uint8_t> trx;
+            {
+              VPackBuilder builder(trx);
+              arangodb::agency::envelope<VPackBuilder>::create(builder)
+                .write()
+                  .remove(rootPath->plan()->coordinators()->server(ctx->server)->str())
+                  .remove(rootPath->plan()->dBServers()->server(ctx->server)->str())
+                  .remove(rootPath->current()->serversRegistered()->server(ctx->server)->str())
+                  .remove(rootPath->current()->dBServers()->server(ctx->server)->str())
+                  .remove(rootPath->supervision()->health()->server(ctx->server)->str())
+                  .remove(rootPath->target()->mapUniqueToShortID()->server(ctx->server)->str())
+                  .remove(rootPath->current()->serversKnown()->server(ctx->server)->str())
+                  .set(rootPath->target()->removedServers()->server(ctx->server)->str(),
+                    timepointToString(std::chrono::system_clock::now()))
+                .precs()
+                  .isEqual(rootPath->supervision()->health()->server(ctx->server)->status()->str(), "FAILED")
+                  .isEmpty(rootPath->supervision()->dbServers()->server(ctx->server)->str())
+                  .isEqual(planVersionPath->str(), agency.get(planVersionPath->vec()))
+                .done()
+              .done();
+            }
+
+            return AsyncAgencyComm().sendWriteTransaction(20s, std::move(trx)).thenValue(
+              [this, ctx = std::move(ctx)](AsyncAgencyCommResult &&result) mutable {
```

Comment:
```suggestion
              [this, ctx = std::move(ctx)](AsyncAgencyCommResult&& result) mutable {
```
do we need to capture `self` as well?

Reply:
This lambda is executed (or destroyed) strictly before the `thenFinal` lambda in `waitForFuture` which captures `self`.